Astro KPO cleanup (#3505)
astronomer/docs: learn/kubepod-operator.md

The components of the connection can also be set or overwritten at the task level by using the arguments `config_file` (to specify the path to the `KubeConfig` file) and `cluster_context`. Setting these parameters in `airflow.cfg` has been deprecated.

:::info Launching Pods in external clusters

If some of your tasks require specific resources such as a GPU, you might want to run them in a different cluster than your Airflow instance. 

> The way that you connect to an external cluster will vary based on where your cluster is hosted and where your Airflow environment is hosted, but generally the following all need to be true to launch a Pod in an external cluster:

Author: ```suggestion
The way that you connect to an external cluster will vary based on where your cluster is hosted and where your Airflow environment is hosted, but generally the following conditions must be met to launch a Pod in an external cluster:
```

4. Make sure your default credentials include a valid and active key for your username. See [Programmatic access](https://docs.aws.amazon.com/general/latest/gr/aws-sec-cred-types.html#access-keys-and-secret-access-keys).

5. Make a copy of `~/.aws/credentials` and `~/.aws/config` available to your Airflow environment. If you run Airflow using the Astro CLI, see [Authenticate to cloud services](https://docs.astronomer.io/astro/cli/authenticate-to-clouds?tab=aws#prerequisites) for steps to mount these credentials to your local Airflow environment.

> 6. [Create a new EKS cluster](https://docs.aws.amazon.com/eks/latest/userguide/create-cluster.html) and assign the newly created role to it.

Author: @manmeetkaur In the original issue you said that " the DAG in that Learn guide is not working because of change of the Class Names". Can you be more specific? For reference: https://github.com/astronomer/docs/blob/main/code-samples/dags/kubepod-operator/kpo_separate_cluster_example.py

Reviewer: Sorry @jwitz I should have been more clear! Basically, if you notice in [these lines](https://github.com/astronomer/docs/blob/main/code-samples/dags/kubepod-operator/kpo_separate_cluster_example.py#L12-L21), the EKS is all caps, but the new class name is "Eks" . See the [Airflow code as reference](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/operators/eks.py).

Reviewer: Also, I would remove [this line](https://github.com/astronomer/docs/blob/main/code-samples/dags/kubepod-operator/kpo_separate_cluster_example.py#L86) . because if we include this, then so many pods will be left in the target cluster. this might be good for debugging, but we should remove from example.

And [this line](https://github.com/astronomer/docs/blob/main/code-samples/dags/kubepod-operator/kpo_separate_cluster_example.py#L80) should be the namespace of the target cluster where you want to run your pods.
